### PR TITLE
cobang: add support for aarch64-linux

### PR DESCRIPTION
**Pull request by dvaerum**
## Description of changes

Added `"aarch64-linux"` to `meta.platforms`

## Things done

I have built cobang on the architecture `aarch64-linux` and it works for me, so can we add support for the `aarch64-linux` architecture?

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/cobang/default.nix
Comment (superherointj) [inline pkgs/applications/misc/cobang/default.nix:102]:
```suggestion
    platforms = lib.platforms.linux;
```

Comment (Aleksanaa) [reply pkgs/applications/misc/cobang/default.nix:102]: Yeah, just assume it works, since it's python

File: pkgs/applications/misc/cobang/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/cobang/default.nix:100]: Would you mind adding yourself to maintainers? Or you can add me to maintainers

File: pkgs/applications/misc/cobang/default.nix
Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix]: I have updated the pull request to use `lib.platform.linux` instead

Comment (superherointj) [reply pkgs/applications/misc/cobang/default.nix]: > `lib.platform.linux`

image

**Comment by superherointj** [comment]
Also [always] squash commits. (It should be a single commit for this change.)

File: pkgs/applications/misc/cobang/default.nix
Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix]: Thanks @superherointj, for noticing that I forgot a `s` in `platform` to should have been fixed

File: pkgs/applications/misc/cobang/default.nix
Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix:100]: I will try to add myself (and you) as maintainers, I just need to get myself added to `maintainers/maintainer-list.nix`.
I have made a https://github.com/NixOS/nixpkgs/pull/331326 which I am waiting on

Comment (superherointj) [reply pkgs/applications/misc/cobang/default.nix:100]: Cherry-pick that commit here. And respect order of commits.

**Comment by superherointj** [comment]
ping @dvaerum

File: pkgs/applications/misc/cobang/default.nix
Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix]: @superherointj I am trying my best to respect the project, I read the document about how to add oneself as a maintainer, but there is a lot left out, like what the subject of the pull request should be. Maybe it was written somewhere else, but when I did not find it.

**Comment by superherointj** [comment]
Please only add clean commits. By clean I mean, no:

> Merge branch 'NixOS:master' into patch-1

First commit should be:
> maintainers: add dvaerum

Order matters because of bisecting.

Second commit should be:
> cobang: add support aarch64-linux
> cobang: add dvaerum as maintainer

You could make the `cobang` change a single commit. But then, you have to add to the body of commit message what you are doing. You are doing two actions. So you comment that.

Options are: 3 commits or 2 commits (adding information to body of commit). Up to you.

File: pkgs/applications/misc/cobang/default.nix
Comment (superherointj) [reply pkgs/applications/misc/cobang/default.nix]: > @superherointj I am trying my best to respect the project, I read the document about how to add oneself as a maintainer, but there is a lot left out, like what the subject of the pull request should be. Maybe it was written somewhere else, but when I did not find it.

**Familiarity**. You cannot guess right what you don't know. But yes, documentation could be better, but realize, different people are at different level at different matters, so there isn't a documentation that fits everyone needs. Some know git or GitHub well, others don't... Most conventions are straightforward. If you want to change something, open a PR, reviewers will debate it [but is a tiring process].

In truth, everything sucks, proposing code sucks, reviewing sucks, committing sucks even more. (Having to own shit sucks.) This is unpaid work, right?

I ask myself everyday why I'm still doing this. Idk. I honestly don't know. =]

If a problem arise, we'll figure out a solution. Don't worry.

Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix]: @superherointj thanks for the link (I will read it in more details and may have follow-up questions), I think that there probably should be a link to the conventions in the document
https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
Because it was there I reading about how to add myself as an maintainer.

Comment (superherointj) [reply pkgs/applications/misc/cobang/default.nix]: I agree that documentation could be improved.

**Review by superherointj — APPROVED** [review]

File: pkgs/applications/misc/cobang/default.nix
Comment (dvaerum) [reply pkgs/applications/misc/cobang/default.nix]: @Aleksanaa 
Okay, I should have fixed the order of the commits (and the commit text there I added myself to the maintainer list)

@superherointj 
I hope everything looks good now

File: pkgs/applications/misc/cobang/default.nix
Comment (superherointj) [inline pkgs/applications/misc/cobang/default.nix:100]:
```suggestion
    maintainers = with maintainers; [ dvaerum aleksana ];
```
I'd default to alphabetical order. While the list may be small now, it could grow in the future.

**Comment by superherointj** [comment]
On commit message:

Things I'd do differently:
- Avoid special characters such as ``. We never know how this could affect something.
- Keep comment objective and concise. For example:
  - cobang: add aarch64-linux platform
  - cobang: add dvaerum as maintainer

This is not stopping this PR from being merged. I'm waiting CI atm.

**Comment by superherointj** [comment]
image

Is this package building for you?

**Comment by superherointj** [comment]
#330241

**Comment by dvaerum** [comment]
> image
> 
> Is this package building for you?

No, not after the 6 of July I think it was, it seems to be a problem with a unit test which are triggered, I have an `overriteAttrs` which disables `nativeCheckInputs` so that the build will work.
```nix
python-zbar_p = pkgs.python312Packages.python-zbar.overrideAttrs (oldAttrs: rec {
  nativeCheckInputs = [ ];
});
```
I have not figured out yet which unittest there needs to be disabled for the build to work, but `cobang` works on my PC.
It was something I wanted to look into it more after this merge request and try to make a pull request with a fix and maybe become a maintainer of that package too, since it does not have one currently

**Comment by superherointj** [comment]
> I have not figured out yet which unittest there needs to be disabled for the build to work, 

I just gave you the solution.

> but `cobang` works on my PC.

The reference must be nixpkgs master. And it is failing at nixpkgs master. While I understand the problem was not introduced by you, and the changes you are proposing are unrelated. We should be fixing it. And I already gave you the solution.

> It was something I wanted to look into it more after this merge request (made it before I learned that the `python-zbar` package broke) and try to make a pull request with a fix and maybe become a maintainer of that package too, since it does not have one currently

Let's do this in this PR since I'm already reviewing this. (You are saving yourself time.)

**Comment by dvaerum** [comment]
> To fix it, I suggest you remove from:
> 
> > pkgs/development/python-modules/python-zbar/default.nix
> 
> The lines:
> 
> ```
>   # distutils usage in setup.py
>   disabled = pythonAtLeast "3.12";
> ```
> 
> Then, it builds fine.
> 
> This also means an extra commit:
> 
>     * `python312Packages.python-zbar: fix build`

I hope I understood your instructions correctly 😇

**Comment by dvaerum** [comment]
> I hope I understood your instructions correctly 😇

Looks like I misunderstood something or there is another problem :confused:

**Comment by superherointj** [comment]
Remove `pythonAtLeast` reference since it is no longer used.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 326368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>cobang</li>
    <li>cobang.dist</li>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>

**Comment by superherointj** [comment]
ping @dvaerum

**Comment by dvaerum** [comment]
Hey @superherointj, thanks for the pointers

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 326368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 326368` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>cobang</li>
    <li>cobang.dist</li>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>

**Pull request merged**
